### PR TITLE
explicitly always prefer latest OSBAPI Version

### LINKED ISSUE #982: Add API surface for broker API version and alpha OSB feature support
Split out from #971 

There should be API fields for setting the version of the OSB API that a broker supports and whether that broker allows alpha features to be enabled.  Note that alpha feature support is probably only meaningful for brokers that work with the most recent version of the API.  The OSB client already supports pinning to a version and has an alpha features flag, so this should just be a matter of adding a couple fields to the Broker's spec:

```go
type BrokerSpec struct {
  // other fields omitted
  OSBSettings *OSBSettings `json:"osbSettings,omitempty"`
}

type OSBSettings struct {
  // APIVersion is the version of the OSB API to use to communicate with this broker.
  // If empty, the latest version of the API will be used.
  APIVersion string `json:"apiVersion,omitempty"`

  // AlphaFeatures characterizes the use of alpha features for this broker.  Unless
  // the value of this field is "Enabled", alpha features are disabled.
  AlphaFeatures *OSBAlphaFeatures `json:"alphaFeatures,omitempty"`
}

type OSBAlphaFeatures string

const (
  OSBAlphaEnabled OSBAlphaFeatures = "Enabled"
  OSBAlphaDisabled OSBAlphaFeatures = "Disabled"
)
```

### PR DESCRIPTION
**Pull request by MHBauer**
- resolves #982 

As agreed in 2017-08-15 meeting.

### CODE REVIEW

File: cmd/controller-manager/app/options/options.go [hunk 1]
```diff
@@ -50,7 +50,7 @@ const (
 	defaultLeaderElectionNamespace      = "kube-system"
 )
 
-var defaultOSBAPIPreferredVersion = osb.Version2_12().HeaderValue()
+var defaultOSBAPIPreferredVersion = osb.LatestAPIVersion().HeaderValue()
```
Comment (duglin) [inline cmd/controller-manager/app/options/options.go]: minor thing but I don't want to mislead people, we should just call this "OSBAPIVersion"

Comment (MHBauer) [inline cmd/controller-manager/app/options/options.go]: going along with all the other flag defaults. would be const, but can't have a const result of a function call.

**Comment by duglin** [comment]
LGTM

**Comment by arschles** [comment]
Travis is backed up again, but @MHBauer has set it up to run against his branch (see [here](https://travis-ci.org/MHBauer/service-catalog/builds/265326367)). Since that run passes, I am going to merge this PR as well.

**Pull request merged**